source/common/router/router.cc

```diff
@@ -198,6 +181,8 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
 
   downstream_headers_ = &headers;
 
+  grpc_request_ = Grpc::Common::hasGrpcContentType(headers);
```

Comment:
I wonder, should we have this be a callback API so the filter doesn't have to recompute/track this? We are making gRPC a fundamental part of request processing at this point which IMO makes sense.

Reply:
Can you add a TODO here about possibly adding a filter API for this?